### PR TITLE
Add SSH keys

### PR DESCRIPTION
**Pull request by huwshimi**
- Add a form to add SSH keys. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1504. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1503. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1505.

QA:
- Navigate to the SSH key list in your profile.
- Click "Import SSH key".
- Try importing a GitHub and Launchpad key and paste a local key.
- Each key should be created and appear in the list.

### CODE REVIEW

File: ui/src/app/preferences/actions/sshkey/sshkey.js [hunk 1]
```diff
@@ -8,4 +8,30 @@ sshkey.fetch = () => ({
...
+  meta: {
+    model: "sshkey",
+    method: "import_keys"
+  },
+  payload: {
+    params
+  }
+});
+
+sshkey.cleanup = params => ({
```
Comment (Caleb-Ellis) [inline ui/src/app/preferences/actions/sshkey/sshkey.js]: Unnecessary params here

File: ui/src/app/preferences/views/SSHKeys/AddSSHKey/AddSSHKey.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  useEffect(() => {
+    return () => {
+      // Clean up saved and error states on unmount.
+      dispatch(sshkeyActions.cleanup());
+    };
+  }, [dispatch]);
+
+  if (saved) {
+    // The snippet was successfully created/updated so redirect to the list.
```
Comment (Caleb-Ellis) [inline ui/src/app/preferences/views/SSHKeys/AddSSHKey/AddSSHKey.js]: Snippet?

File: ui/src/app/preferences/views/SSHKeys/SSHKeyFormFields/SSHKeyFormFields.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import Textarea from "app/base/components/Textarea";
+
+export const SSHKeyFormFields = ({ editing, formikProps }) => {
+  const saving = useSelector(sshkeySelectors.saving);
+  const saved = useSelector(sshkeySelectors.saved);
+  const errors = useSelector(sshkeySelectors.errors);
+  useFormikErrors(errors, formikProps);
+  let sshkeyErrors;
+  if (errors) {
+    if (typeof sslkeyErrors === "string") {
```
Comment (Caleb-Ellis) [inline ui/src/app/preferences/views/SSHKeys/SSHKeyFormFields/SSHKeyFormFields.js]: This should be `typeof errors === "string"`

File: ui/src/app/preferences/actions/sshkey/sshkey.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/preferences/actions/sshkey/sshkey.js]: Oops.

File: ui/src/app/preferences/views/SSHKeys/AddSSHKey/AddSSHKey.js (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/preferences/views/SSHKeys/AddSSHKey/AddSSHKey.js]: Oops.

File: ui/src/app/preferences/views/SSHKeys/SSHKeyFormFields/SSHKeyFormFields.js (hunk 3 above)
Comment (huwshimi) [reply ui/src/app/preferences/views/SSHKeys/SSHKeyFormFields/SSHKeyFormFields.js]: Oops.

**Comment by huwshimi** [comment]
@Caleb-Ellis I've filed https://github.com/canonical-web-and-design/maas-ui/issues/243 for the border issue.

I think an issue for each epic is OK, I'll add one for prefs.

**Pull request merged**